exercises/complex-numbers/test/test_complex_numbers.c

```diff
@@ -0,0 +1,192 @@
+        { { .real = 1.0, .imag = 2.0 }, { .real = 3.0, .imag = 4.0 }, { .real = 4.0, .imag = 6.0 } }
+    };
+
+    unsigned int i;
+    for (i = 0; i < sizeof(cases) / sizeof(Complex[3]); i++) {
+        Complex result = c_add(&cases[i][0], &cases[i][1]);
+        
+        TEST_ASSERT_EQUAL(result.real, cases[i][2].real);
+        TEST_ASSERT_EQUAL(result.imag, cases[i][2].imag);
+    }
+}
+
+// subtraction
+void test_subtraction(void) {
+    Complex cases[][3] = {
+        { { .real = 1.0, .imag = 0.0 }, { .real = 2.0, .imag = 0.0 }, { .real = -1.0, .imag = 0.0 } },
+        { { .real = 0.0, .imag = 1.0 }, { .real = 0.0, .imag = 2.0 }, { .real = 0.0, .imag = -1.0 } },
+        { { .real = 1.0, .imag = 2.0 }, { .real = 3.0, .imag = 4.0 }, { .real = -2.0, .imag = -2.0 } }
+    };
+    
+    unsigned int i;
+    for (i = 0; i < sizeof(cases) / sizeof(Complex[3]); i++) {
+        Complex result = c_sub(&cases[i][0], &cases[i][1]);
+
+        TEST_ASSERT_EQUAL(result.real, cases[i][2].real);
+        TEST_ASSERT_EQUAL(result.imag, cases[i][2].imag); 
+    }
+}
+
+// multiplication
+void test_multiplication(void) {
+    Complex cases[][3] = {
+        { { .real = 1.0, .imag = 0.0 }, { .real = 2.0, .imag = 0.0 }, { .real = 2.0, .imag = 0.0 } },
+        { { .real = 0.0, .imag = 1.0 }, { .real = 0.0, .imag = 2.0 }, { .real = -2.0, .imag = 0.0 } },
+        { { .real = 1.0, .imag = 2.0 }, { .real = 3.0, .imag = 4.0 }, { .real = -5.0, .imag = 10.0 } }
+    };
+    
+    unsigned int i;
+    for (i = 0; i < sizeof(cases) / sizeof(Complex[3]); i++) {
+        Complex result = c_mul(&cases[i][0], &cases[i][1]);
+
+        TEST_ASSERT_EQUAL(result.real, cases[i][2].real);
+        TEST_ASSERT_EQUAL(result.imag, cases[i][2].imag); 
+    }
+}
+
+// division
+void test_division(void) {
+    Complex cases[][3] = {
+        { { .real = 1.0, .imag = 0.0 }, { .real = 2.0, .imag = 0.0 }, { .real = 0.5, .imag = 0.0 } },
+        { { .real = 0.0, .imag = 1.0 }, { .real = 0.0, .imag = 2.0 }, { .real = 0.5, .imag = 0.0 } },
+        { { .real = 1.0, .imag = 2.0 }, { .real = 3.0, .imag = 4.0 }, { .real = 0.44, .imag = 0.08 } }
+    };
+    
+    unsigned int i;
+    for (i = 0; i < sizeof(cases) / sizeof(Complex[3]); i++) {
+        Complex result = c_div(&cases[i][0], &cases[i][1]);
+
+        TEST_ASSERT_EQUAL(result.real, cases[i][2].real);
+        TEST_ASSERT_EQUAL(result.imag, cases[i][2].imag); 
+    }
+}
+
+// absolute value
+void test_absolute_value(void) {
+    Complex cases[] = {
+        { .real = 5.0, .imag = 0.0 },
+        { .real = -5.0, .imag = 0.0 },
+        { .real = 0.0, .imag = 5.0 },
+        { .real = 0.0, .imag = -5.0 },
+        { .real = 3.0, .imag = 4.0 }
+    };
+
+    unsigned int i;
+    for (i = 0; i < sizeof(cases) / sizeof(Complex); i++) {
+        double result = c_abs(&cases[i]);
+
+        TEST_ASSERT_EQUAL(5.0, result);
+    }
+}
+
+// conjugate
```

Comment:
These comments don't seem to add anything that the test name doesn't already say. Can they be removed?

Reply:
I'll get rid of them.